src/main/java/dinkplugin/message/DiscordMessageHandler.java

```diff
@@ -33,14 +34,34 @@
 
 @Slf4j
 @Singleton
-@RequiredArgsConstructor(onConstructor_ = { @Inject })
 public class DiscordMessageHandler {
     private final Client client;
     private final DrawManager drawManager;
     private final OkHttpClient httpClient;
     private final DinkPluginConfig config;
     private final ScheduledExecutorService executor;
 
+    @Inject
+    public DiscordMessageHandler(Client client, DrawManager drawManager, OkHttpClient httpClient, DinkPluginConfig config, ScheduledExecutorService executor) {
+        this.client = client;
+        this.drawManager = drawManager;
+        this.config = config;
+        this.executor = executor;
+        this.httpClient = httpClient.newBuilder()
+            .addInterceptor(chain -> {
+                Request request = chain.request().newBuilder()
+                    .header("User-Agent", DinkPlugin.USER_AGENT)
+                    .build();
+                Interceptor.Chain updatedChain = chain;
+                // Allow longer timeout when writing a screenshot file to overcome slow internet speeds
+                if (request.body() instanceof MultipartBody && Utils.hasImage((MultipartBody) request.body())) {
+                    updatedChain = chain.withWriteTimeout(30, TimeUnit.SECONDS); // default is 10 seconds
```

Comment:
Should we get the timeout seconds from a hidden config setting instead?

Reply:
Yeah I think that would good, it adds some automatic documentation for the values floating around